Solution1/authentication_app/Program.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+
+// Пользователю предлагается ввести имя пользователя и пароль, при совпадении имени пользователя
+// и пароля с имеющимися в системе – выдать сообщение об успешном логине, при несовпадении –
+// выдать сообщение об ошибке и предложить еще одну попытку. 
+// Максимальное количество попыток – 3, при 3 неуспешных попытках авторизации – выдать сообщение об
+// ошибке, закрыть приложение после нажатия любого символа.
+
+namespace AuthenticationApp
+{
+    class Program
+    {
+        static bool CheckAuthentication()
+        {
+            string user = "admin";
+            string password = "pass";
+
+            Console.WriteLine("Enter user name:");
+            string userName = Console.ReadLine();
+
+            Console.WriteLine("Enter password:");
+            string userPass = Console.ReadLine();
+
+            bool credMatch = (userName == user) && (userPass == password);
+
+            return credMatch;
+        }
+
+        static void Main()
+        {
+            int i = 0;
+            do
+            {
+                if (CheckAuthentication() == true)
```

Comment:
You could remove check == true, because you are checking result of the function already, so for now it looks like this:
if (true == true) if CheckAuthentication() return true
and if (false == true) if false, so first exp will be always true and second will be false
so you could have just if(CheckAuthentication())

Reply:
Done